tests/parsers/test_cp/test_cp_default_6_5_autopilot_.yml

```diff
@@ -0,0 +1,188 @@
+parameters:
+  bands:
+    bands:
+    - - - -12.075352784575822
+        - -6.283166641110415
+        - -4.026362047368278
+        - -3.175073740386193
+    bands_units: eV
+    occupations:
+    - - - 1.0
+        - 1.0
+        - 1.0
+        - 1.0
+  beta_real_space: false
+  charge_density: ./charge-density.dat
+  conjugate_gradient: true
+  constraint_mag: 0
+  creator_name: cp
+  creator_version: '6.5'
+  dft_exchange_correlation: null
+  do_magnetization: false
+  do_not_use_time_reversal: null
+  exit_status: 0
+  fermi_energy_units: eV
+  fft_grid:
+  - 60
+  - 60
+  - 60
+  format_name: QEXSD
+  format_version: 19.03.04
+  has_dipole_correction: false
+  has_electric_field: false
+  inversion_symmetry: false
+  k_points:
+  - - 0.0
+    - 0.0
+    - 0.0
+  k_points_units: 1 / angstrom
+  k_points_weights:
+  - 2.0
+  last_successful_printout: 30
+  lattice_symmetries: []
+  lda_plus_u_calculation: false
+  lkpoint_dir: false
+  lsda: false
+  magnetization_angle1:
+  - 0.0
+  - 0.0
+  magnetization_angle2:
+  - 0.0
+  - 0.0
+  no_time_rev_operations: null
+  non_colinear_calculation: false
+  number_of_atomic_wfc: 6
+  number_of_bands: 4
+  number_of_bravais_symmetries: null
+  number_of_electrons: 8.0
+  number_of_k_points: 1
+  number_of_spin_components: 1
+  number_of_symmetries: null
+  q_real_space: false
+  rho_cutoff: 4353.821352096
+  rho_cutoff_units: eV
+  smooth_fft_grid:
+  - 60
+  - 60
+  - 60
+  spin_orbit_calculation: false
+  spin_orbit_domag: false
+  starting_magnetization:
+  - 0.0
+  - 0.0
+  structure:
+    atomic_positions_units: Angstrom
+    atoms:
+    - - H
+      - - 0.3077936327955275
+        - 0.49074731600773003
+        - 0.6020237422809962
+    - - H
+      - - -0.25285429616438865
+        - -0.2184308056415271
+        - -0.7702509858867927
+    - - O
+      - - -0.16627752338900367
+        - -0.26930737986161374
+        - 0.2154517857488999
+    cell:
+      atoms:
+      - - H
+        - - 0.3077936327955275
+          - 0.49074731600773003
+          - 0.6020237422809962
+      - - H
+        - - -0.25285429616438865
+          - -0.2184308056415271
+          - -0.7702509858867927
+      - - O
+        - - -0.16627752338900367
+          - -0.26930737986161374
+          - 0.2154517857488999
+      lattice_vectors:
+      - - 5.2917720859
+        - 0.0
+        - 0.0
+      - - 0.0
+        - 5.2917720859
+        - 0.0
+      - - 0.0
+        - 0.0
+        - 5.2917720859
+      volume: 148.1847095290449
```

Comment:
This was always the case, I don't think that I touched that part of the plugin! The only new feature that I coded was the autopilot input.

Reply:
I think it is just because of the new XML. The old XML did not include the structure and bands, or at least the old parser did not parse them and put them in the `output_parameters`. We can see this from the existing `default` parser test, where the `output_parameters` do not include these quantities. Anyway, we had the same problem when we first added support for the new XML for the `PwParser` and we spend a great deal of time to make sure that the keys in `output_parameters` would remain the same. This is why we added this big mapping function in `parse_xml/pw/parse.py` where the data parsed automatically from the XML using the schema library, is mapped onto different keys that were also produced by the legacy parser. @giovannipizzi this is clearly not done here. Fine to keep it as is, accepting that the structure of `output_parameters` will depend heavily on the version of the XML output file?

Reply:
Hi, it's a good point - ideally I would suggest to do what we did in pw, but in practice this is probably just too much work and I don't know if we have anyone who would have the time to do it; in the end, I think it's better to have a plugin that is usable even if the content of the output dictionary has changed (at the moment I think it's not working with the new XML). 

The only thing that I don't know if it's worth trying to copy-paste the mapping function, trying to apply to CP and do the minimal modifications to make it work for CP. But if it's too complex, better to merge this.

Regarding bands and structure - I think it's OK to remove them, especially if they weren't there before.
For the structure, this should be in general taken from the last snapshot of the trajectory.
Not sure about the bands, but these should be probably in one of the output arrays? In any case, I agree that if we really want to keep them we should put them in a BandsData.
And if the information is in the XML that is anyway stored in AiiDA, these can always be parsed later, and we can extend later the parser to also output those in a backward-compatible way.

Reply:
Isn't at this moment the xml parser doing exactly the same both for cp and pw? If I understood correctly what you are saying, the dictionary that you are talking about is https://github.com/rikigigi/aiida-quantumespresso/blob/baeec661bf03a4b7c53c3ae34f4ff7caad62d350/aiida_quantumespresso/parsers/parse_xml/parse.py#L200-L253 , and this code is called by cp's parser also. Regarding the legacy cp and pw parser, I didn't touched them